python/pyspark/sql/functions.py

```diff
@@ -2105,6 +2105,14 @@ def udf(f=None, returnType=StringType()):
     >>> import random
     >>> random_udf = udf(lambda: int(random.random() * 100), IntegerType()).asNondeterministic()
 
+    .. note:: The user-defined functions are considered to be able to return null values by default.
+        If your function is not nullable, call `asNonNullable` on the user defined function.
+        E.g.:
+
+    >>> from pyspark.sql.types import StringType
+    >>> import getpass
+    >>> getuser_udf = udf(lambda: getpass.getuser(), StringType()).asNonNullable()
```

Comment:
Default is string. I think we can omit `StingType`:

```python
getuser_udf = udf(lambda: getpass.getuser(), "string").asNonNullable()
```

Reply:
Why? Use of StringType() is consistent with other tests.

Reply:
Oh, simply because it's shorter :-). That's fine if you prefer this way. I don't feel strongly.